lib/ansible/executor/task_executor.py

```diff
@@ -488,6 +488,12 @@ def _execute(self, variables=None):
 
         self._handler = self._get_action_handler(connection=self._connection, templar=templar)
 
+        # Apply default params for action
+        if self._play_context.module_defaults is not None and self._task.action in self._play_context.module_defaults:
+            tmp_args = self._play_context.module_defaults[self._task.action].copy()
+            tmp_args.update(self._task.args)
+            self._task.args = tmp_args
```

Comment:
That merges the args in the wrong order. That applies the defaults on top of the user-specified params.